android/src/com/mapswithme/maps/editor/LanguagesFragment.java

```diff
@@ -15,14 +23,30 @@
   @Override
   protected RecyclerView.Adapter createAdapter()
   {
-    return new LanguagesAdapter(this, Editor.nativeGetSupportedLanguages());
+    Bundle args = getArguments();
+    HashSet<String> existingLanguages = new HashSet<>(Arrays.asList(args.getStringArray(EditorHostFragment.kExistingLocalizedNames)));
```

Comment:
So many transformations of string array - to list, then to hash set. Why not to instantiate an empty hash set and then add to it all strings from kExistingLocalizedNames?